File: ulyp-ui/src/main/java/com/ulyp/ui/renderers/RenderedNotRecordedObject.java
package com.ulyp.ui.renderers;

import com.ulyp.core.Type;
import com.ulyp.ui.RenderSettings;
import javafx.scene.text.Text;

public class RenderedNotRecordedObject extends RenderedObject {

    protected RenderedNotRecordedObject(RenderSettings renderSettings) {
        super(Type.unknown());

        if (renderSettings.showTypes()) {
            super.getChildren().add(new Text("Method has not yet returned any value: ?"));
        } else {
            super.getChildren().add(new Text("?"));
        }
    }
}

File: ulyp-ui/src/main/kotlin/com/ulyp/ui/renderers/RenderedNotRecordedObject.kt
package com.ulyp.ui.renderers

import com.ulyp.core.Type
import com.ulyp.ui.RenderSettings
import com.ulyp.ui.renderers.RenderedObject
import javafx.scene.text.Text

class RenderedNotRecordedObject(renderSettings: RenderSettings) : RenderedObject(Type.unknown()) {
    init {
        if (renderSettings.showTypes()) {
            super.getChildren().add(Text("Method has not yet returned any value: ?"))
        } else {
            super.getChildren().add(Text("?"))
        }
    }
}